solver/src/main/java/org/chocosolver/solver/ModelAnalyser.java

```diff
@@ -0,0 +1,494 @@
+            for (int i = 0; i < list.size(); i++) {
+                String propType = list.get(i);
+                ConstraintTypeStatistics cstrTypeStats = createConstraintTypeStatistics(cstrType, propType);
+                ps.println(cstrTypeStats.toString(true, false, true));
+                if (i + 1 < list.size()) {
+                    ps.println();
+                }
+            }
+            ps.println();
+            ps.println();
+        }
+        ps.println("################################################");
+        ps.println("########## END OF CONSTRAINTS ANALYSIS #########");
+        ps.println("################################################");
+    }
+
+    public void printAnalysis() {
+        printAnalysis(System.out);
+    }
+
+    public void printAnalysis(PrintStream ps) {
+        ps.println();
+        ps.println("################################################################################################");
+        ps.println("#################################### BEGIN OF MODEL ANALYSIS ###################################");
+        ps.println("################################################################################################");
+        ps.println();
+        printVariableAnalysis(ps);
+        ps.println();
+        ps.println();
+        printConstraintAnalysis(ps);
+        ps.println();
+        ps.println("################################################################################################");
+        ps.println("##################################### END OF MODEL ANALYSIS ####################################");
+        ps.println("################################################################################################");
+        ps.println();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////     JSON EXPORT     ////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    private static void toFile(String path, Object toWrite) {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            File f = new File(path);
+            if(!f.getParentFile().exists()) {
+                f.getParentFile().mkdirs();
+            }
+            DefaultPrettyPrinter.Indenter indenter = new DefaultIndenter("    ", DefaultIndenter.SYS_LF);
+            DefaultPrettyPrinter printer = new DefaultPrettyPrinter();
+            printer.indentObjectsWith(indenter);
+            printer.indentArraysWith(indenter);
+            String s = mapper.writer(printer).writeValueAsString(toWrite);
+            FileWriter fw = new FileWriter(path);
+            fw.write(s);
+            fw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static class ModelAnalyserJson {
```

Comment:
I would not use "Json" in the class name. 
You can serialize it into json, but you don't have to. 

Also maybe ModelAnalysis is more appropriate than ModelAnalyser

Reply:
Indeed, ModelAnalysis is a more appropriate name for this class. I am changing it right away.